### PR TITLE
Hide fields for case triggered plan like hidden in routine plan

### PR DESCRIPTION
**Pull request by haseemVD**
This PR addresses this [part of issue](https://github.com/onaio/reveal-frontend/issues/1378#issuecomment-773263684)

### CODE REVIEW

File: src/containers/pages/InterventionPlan/UpdatePlan/index.tsx [hunk 1]
```diff
@@ -138,10 +139,20 @@ const UpdatePlan = (props: RouteComponentProps<RouteParams> & UpdatePlanProps) =
...
     initialValues.fiReason === CASE_TRIGGERED;
+
+  const isCaseTriggeredOrRoutinePlan =
+    CASE_TRIGGERED_DRAFT_EDIT_ADD_ACTIVITIES &&
+    (planStatus === PlanStatus.DRAFT ||
+      planStatus === PlanStatus.ACTIVE ||
+      planStatus === PlanStatus.COMPLETE) &&
+    isFIOrDynamicFI(initialValues.interventionType) &&
+    initialValues.fiReason !== CASE_TRIGGERED &&
+    initialValues.fiReason !== ROUTINE;
```
Comment (ciremusyoka) [inline src/containers/pages/InterventionPlan/UpdatePlan/index.tsx:150]: Do we need this?
Also i think this condition will always be false. Right?

Comment (haseemVD) [reply src/containers/pages/InterventionPlan/UpdatePlan/index.tsx]: Checks updated.

File: src/containers/pages/InterventionPlan/UpdatePlan/tests/index.test.tsx [hunk 2]
```diff
@@ -927,4 +927,59 @@ describe('components/InterventionPlan/UpdatePlan', () => {
     // check submit button - should not be disabled
     expect(wrapper.find('#planform-submit-button button').prop('disabled')).toEqual(false);
   });
+
+  it('should pass correct array to hide fields for case triggered draft plans', async () => {
```
Comment (ciremusyoka) [inline src/containers/pages/InterventionPlan/UpdatePlan/tests/index.test.tsx:931]: let's have additional tests here showing those fields are hidden on the form as well.

Comment (haseemVD) [reply src/containers/pages/InterventionPlan/UpdatePlan/tests/index.test.tsx:931]: Added tests for some fields, to checks if they are hidden or not.

**Review by ciremusyoka — DISMISSED** [review]

File: src/containers/pages/InterventionPlan/UpdatePlan/index.tsx [hunk 3]
```diff
@@ -138,10 +138,11 @@ const UpdatePlan = (props: RouteComponentProps<RouteParams> & UpdatePlanProps) =
     planStatus === PlanStatus.DRAFT &&
     isFIOrDynamicFI(initialValues.interventionType) &&
     initialValues.fiReason === CASE_TRIGGERED;
+
   const hiddenFields =
     HIDE_PLAN_FORM_FIELDS_ON_EDIT.length > 0 &&
     hideFieldsOnPlanStatuses.includes(initialValues.status) &&
-    !isCaseTriggeredAndDraft
+    isFIOrDynamicFI(initialValues.interventionType)
```
Comment (moshthepitt) [inline src/containers/pages/InterventionPlan/UpdatePlan/index.tsx:145]: Seems like this is the only change.  Could you describe how it solves the issue?

Comment (ciremusyoka) [reply src/containers/pages/InterventionPlan/UpdatePlan/index.tsx:145]: @moshthepitt this ensures this fields are hidden for all `FI` and `Dynamic FI` either `Routine` or `Case Triggered` plans as opposed to earlier version that was hiding the fields for `Routine`  and not `case triggered` in draft mode.

Comment (moshthepitt) [reply src/containers/pages/InterventionPlan/UpdatePlan/index.tsx:145]: > this ensures this fields are hidden for all FI and Dynamic FI either Routine or Case Triggered plans as opposed to earlier version that was hiding the fields for Routine and not case triggered in draft mode.

Would have been nice to have this in a commit message :)

**Review by moshthepitt — APPROVED** [review]

**Review by moshthepitt — APPROVED** [review]

**Pull request merged**